nrich-notification/README.md

```diff
@@ -33,33 +46,76 @@ To be able to use this module following configuration is required:
+---
 
-When using Spring Web MVC users can use `NotificationResponseService` implementation `WebMvcNotificationResponseService` that is a wrapper around `NotificationResolverService` providing a
-convenient way of returning a notification with a response.
+### [NotificationResolverService][notification-resolver-service-url]
+
+[`NotificationResolverService`][notification-resolver-service-url] is the core service that is responsible for creating notifications.
+It can create notifications for validation errors, exceptions or action names.
+Consequently, it accepts either Spring's `Errors` or Jakarta `ConstraintViolationException` for creating validation notifications, `Throwable` for creating error notifications and `String` for action
+notification.
+
+---
+
+### [NotificationResponseService][notification-response-service-url]
+
+[`NotificationResponseService`][notification-response-service-url] is a service that adds support for additional data in notifications by introducing additional methods that
+return [`NotificationDataResponse`][notification-data-response-url].
+When using Spring Web MVC users can use [`NotificationResponseService`][notification-response-service-url] implementation
+[`WebMvcNotificationResponseService`][web-mvc-notification-response-service-url] that is a wrapper around [`NotificationResolverService`][notification-resolver-service-url] providing a convenient way
+of returning a notification with a response.
+
+---
+
+### Default `MessageSource` setup
+
+`nrich-notification` also includes [`nrich-notification-messages`][nrich-notification-messages-url] property files that contain required default messages and should be added to
+`MessageSource` (**this is done automatically when using Spring Boot starter module**).
 
 ## Usage
 
-`nrich-notification` is used as a dependency of `nrich-webmvc` module. It is used in a `RestControllerAdvice` and on exceptions it creates notifications that are then sent to the client,
-so a common usage would be to use it in a `ControllerAdvice` to have unified error notification handling.
+Following examples of usage demonstrate `nrich-notification` module's ability to create notifications that respond to **actions**, **exceptions** and **validation errors**.
+`nrich-notification` is used as a dependency of [`nrich-webmvc`][nrich-webmvc-url] module and as such is better explained with examples that are based on requests and responses.
+
+### Basic notification structure
+
+Every notification class in `nrich-notification` inherits [`Notification`][notification-url] class that is specified in [`nrich-notification-api`][nrich-notification-api-url] module and has these
+attributes:
+
+| Attribute                | Description                                       |
+|--------------------------|---------------------------------------------------|
+| `title`                  | Notification title.                               |
```

Comment:
Not sure about backticks here. It seems better without.

Reply:
I would disagree here because it stands out better with backtiks, and collegues said they like it better with.

Reply:
Ok, we can leave it like this if the community likes it better.